stdlib/array.mli

```diff
@@ -328,6 +328,17 @@ val fast_sort : ('a -> 'a -> int) -> 'a array -> unit
 (** Same as {!sort} or {!stable_sort}, whichever is
     faster on typical input. *)
 
+val shuffle :
+  rand: (* thwart tools/sync_stdlib_docs *) (int -> int) -> 'a array -> unit
+(** [shuffle rand a] randomly permutes [a]'s elements using [rand]
+    for randomness.
```

Comment:
Something should be said about the function implementation a **uniform** distribution of the permutation (assuming the uniform distribution of `rand`).

Reply:
Is there a reason to use a different word for the function name and in its docstring (shuffle vs permute)?  I'm not against it, just checking that this is intentional.

Reply:
I added a sentence to the doc string regarding the uniform distribution. 

Regarding shuffle vs permute. I'm not attached to the name both seem to be in use. A quick grab bag, it seems containers, [`scala`](https://www.scala-lang.org/api/2.13.5/scala/util/Random$.html#shuffle[T,C](xs:IterableOnce[T])(implicitbf:scala.collection.BuildFrom[xs.type,T,C]):C) [`ruby`](https://apidock.com/ruby/v2_5_5/Array/shuffle),  [C++](https://en.cppreference.com/w/cpp/algorithm/random_shuffle), lots of other libraries in other language use `shuffle` and base uses `permute`. Sherlocode tells me [shuffle](https://sherlocode.com/?q=shuffle) is not uncommon in ocaml code and here are the results for [permute](https://sherlocode.com/?q=permute).

One (rather weak) argument in favor of `shuffle` is that `permute` could be taken as swapping two elements, though in a reading context I don't think it would be. I'd be more in favour of `permutation` if we were not doing it in place.

If people want to debate this to death, please do I will gladly take the name people compromise on.